src/Behaviors/Meshes/baseSixDofDragBehavior.ts

```diff
@@ -0,0 +1,378 @@
+
+    protected _scene: Scene;
+    protected _moving = false;
+    protected _ownerNode: TransformNode;
+    protected _dragging: boolean = false;
+
+    /**
+     * The list of child meshes that can receive drag events
+     * If `null`, all child meshes will receive drag event
+     */
+    public draggableMeshes: Nullable<AbstractMesh[]> = null;
+
+    /**
+     * How much faster the object should move when the controller is moving towards it. This is useful to bring objects that are far away from the user to them faster. Set this to 0 to avoid any speed increase. (Default: 3)
+     */
+    public zDragFactor = 3;
+    /**
+     * The id of the pointer that is currently interacting with the behavior (-1 when no pointer is active)
+     */
+    public get currentDraggingPointerId() {
+        if (this.currentDraggingPointerIds[0] !== undefined) {
+            return this.currentDraggingPointerIds[0];
+        }
+        return -1;
+    }
+
+    public set currentDraggingPointerId(value: number) {
+        this.currentDraggingPointerIds[0] = value;
+    }
+
+    /**
+     * In case of multipointer interaction, all pointer ids currently active are stored here
+     */
+    public currentDraggingPointerIds: number[] = [];
+
+    /**
+     * Get or set the currentDraggingPointerId
+     * @deprecated Please use currentDraggingPointerId instead
+     */
+    public get currentDraggingPointerID(): number {
+        return this.currentDraggingPointerId;
+    }
+    public set currentDraggingPointerID(currentDraggingPointerID: number) {
+        this.currentDraggingPointerId = currentDraggingPointerID;
+    }
+    /**
+    /**
+     * If camera controls should be detached during the drag
+     */
+    public detachCameraControls = true;
+
+    /**
+     * Fires each time a drag starts
+     */
+    public onDragStartObservable = new Observable<{ position: Vector3 }>();
+    /**
+     * Fires each time a drag happens
+     */
+    public onDragObservable = new Observable<{ delta: Vector3; position: Vector3 }>();
+    /**
+     *  Fires each time a drag ends (eg. mouse release after drag)
+     */
+    public onDragEndObservable = new Observable<{}>();
+
+    /**
+     * Should the behavior allow simultaneous pointers to interact with the owner node.
+     */
+    public allowMultiPointer: boolean = true;
+
+    /**
+     *  The name of the behavior
+     */
+    public get name(): string {
+        return "BaseSixDofDrag";
+    }
+
+    /**
+     *  Returns true if the attached mesh is currently moving with this behavior
+     */
+    public get isMoving(): boolean {
+        return this._moving;
+    }
+
+    /**
+     *  Initializes the behavior
+     */
+    public init() {}
+
+    /**
+     * In the case of multiple active cameras, the cameraToUseForPointers should be used if set instead of active camera
+     */
+    private get _pointerCamera() {
+        if (this._scene.cameraToUseForPointers) {

```

Comment:
just a thought - maybe define this as a getter  in the scene instead? so the camera that is set to be the camera to use for pointers will return (if defined), otherwise the scene active camera will return. This way we can stay consistent between components. 
Not sure what the API change requirements here, so - take that as an untested suggestion :-)

Reply:
This was also in `SixDofDragBehavior.ts`, but I agree it should be in the scene as it sound like something we could reuse. But isn't it a problem to add yet another property in the scene ? I think @sebavan mentionned that we shouldn't add too many to not trigger the slow path ?